### PR TITLE
Test-case for API Scopes

### PR DESCRIPTION
**Pull request by sambaheerathan**

### CODE REVIEW

File: modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+        // adding new role subscriber
+        userManagementClient.addRole("subscriber", new String[]{}, new String[]{"/permission/admin/login",
+                                                                                "/permission/admin/manage/api/subscribe"});
+
+        // crating user john
+        userManagementClient.addUser("john", "john123", new String[]{"subscriber"}, "user_role");
+
+
+        // Adding API
```
Comment (nuwand) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: Variable names should start with a simple case letter and follow camel case naming style.

**Comment by sambaheerathan** [comment]
I have completed the mentioned changes.

File: modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                                  "\"method\":\"OPTIONS\",\"parameters\":[]}]}]},\"description\":\"\",\"path\":" +
+                                  "\"/default\"}],\"info\":{\"title\":\"" + API_NAME + "\",\"termsOfServiceUrl\":\"" +
+                                  "\",\"description\":\"\",\"license\":\"\",\"contact\":\"\",\"licenseUrl\":\"\"}}";
+
+
+        apiPublisher.updateResourceOfAPI(API_PROVIDER, API_NAME, API_VERSION, modifiedResource);
+
+        // For Admin user
+        // create new application and subscribing
+        apiStore.login("admin", "admin");
```
Comment (laki88) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: you can get user name password from automation context rather hardcoding

File: modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        response = HttpRequestUtil.doGet(getApiInvocationURLHttp("testScopeAPI/1.0.0/ITEM"), requestHeaders);
+        Assert.assertEquals(response.getResponseCode(), 200, "User John cannot access the GET Method");
+
+        try {
+            // Accessing POST method
+            endPointURL = new URL(getApiInvocationURLHttp("testScopeAPI/1.0.0/PRODUCT/35"));
+            response = HttpRequestUtil.doPost(endPointURL, "<resource><PRICE>8.5</PRICE></resource>", requestHeaders);
+            Assert.assertTrue(response.getResponseCode() != 200, "testRole John can access the POST Method");
+
+        } catch (Exception e) {
```
Comment (nuwand) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: Do not catch generic Exception. Catch and handle the specific exception.

File: modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+            storeURLHttp = getStoreServerURLHttp();
+        }
+
+        apiPublisher = new APIPublisherRestClient(publisherURLHttp);
+
+        apiStore = new APIStoreRestClient(storeURLHttp);
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Testing the scopes with admin, subscriber roles")
+    public void testSetScopeToResourceTestCase() throws Exception {
```
Comment (nuwand) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: Any particular reason for throwing generic Exception? Please remove if possible, comment if an unavoidable reason exists.

File: modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java (hunk 2 above)
Comment (sambaheerathan) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: fixed in https://github.com/wso2/product-apim/pull/173

File: modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java (hunk 3 above)
Comment (sambaheerathan) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: Fixed in https://github.com/wso2/product-apim/pull/173

File: modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java (hunk 4 above)
Comment (sambaheerathan) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: fixed in https://github.com/wso2/product-apim/pull/173

File: modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+    private static final String API_VERSION = "1.0.0";
+
+    private static final String API_PROVIDER = "admin";
+
+    private static final String APP_NAME = "NewApplication";
+
+    private static final String USER_JOHN = "john";
+
+    private static final String SUBSCRIBER_ROLE = "subscriber";
```
Comment (imesh) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: May be its good to remove blank lines between private variables.

Comment (sambaheerathan) [inline modules/integration/tests/src/test/java/org/wso2/carbon/am/tests/sample/APIScopeTestCase.java]: Currently our code is in frozen state due to release i will fix this once its released

**Pull request merged**